pkg/report/report.go

```diff
@@ -383,6 +391,12 @@ func Generate(ctx context.Context, path string, mrs yara.MatchRules, c malconten
 				continue
 			}
 
+			// If we find a match in the map for the metadata key, that's the rule to override
+			var overrideRule string
+			if match, exists := mrsMap[k]; exists {
```

Comment:
This should check that the current rule has an override tag, otherwise we are bound to run into weird edge cases where a rule has a metadata field matching another rule name.

Reply:
Added in [`2fa8a07` (#481)](https://github.com/chainguard-dev/malcontent/pull/481/commits/2fa8a07151d287575f18ef4313e190a93c5b51a8).